src/NHibernate.Test/Linq/ExtensionMethods.cs

```diff
@@ -2,18 +2,24 @@
 
 namespace NHibernate.Test.Linq
 {
-    static class ExtensionMethods
-    {
-        [LinqExtensionMethod("Replace")]
-        public static string ReplaceExtension(this string subject, string search, string replaceWith)
-        {
-            return null;
```

Comment:
I have replaced `null` by `subject?.Replace(search, replaceWith) + " (done in-memory)";` for allowing better asserts. (There were none, the original test was not validating the query was not throwing.)